src/conmon.c

```diff
@@ -547,6 +547,28 @@ static bool has_v1_oom()
 	return false;
 }
 
+/* write the appropriate files to tell the caller there was an oom event
+ * this can be used for v1 and v2 OOMS
+ * Note: the caller is responsible for closing oom_fd
+ */
+static int write_oom_files() {
+	int oom_fd = -1;
+	ninfo("OOM received");
+	if (opt_persist_path) {
+		_cleanup_close_ int ctr_oom_fd = -1;
+		_cleanup_free_ char *ctr_oom_file_path = g_build_filename(opt_persist_path, "oom", NULL);
+		ctr_oom_fd = open(ctr_oom_file_path, O_CREAT, 0666);
+		if (ctr_oom_fd < 0) {
+			nwarn("Failed to write oom file");
+		}
+	}
+	oom_fd = open("oom", O_CREAT, 0666);
+	if (oom_fd < 0) {
+		nwarn("Failed to write oom file");
+	}
+	return oom_fd;
```

Comment:
it could be turned into:
```suggestion
	return oom_fd >= 0 ? 0 : -1;
```